xds/src/main/java/io/grpc/xds/AbstractXdsClient.java

```diff
@@ -330,6 +330,18 @@ private void handleRpcStreamClosed(Status error) {
         return;
       }
 
+      if (responseReceived || retryBackoffPolicy == null) {
+        // Reset the backoff sequence if had received a response, or backoff sequence
+        // has never been initialized.
+        retryBackoffPolicy = backoffPolicyProvider.get();
+      }
+      // Need this here to avoid tsan race condition in XdsClientImplTestBase.sendToNonexistentHost
+      long elapsed = stopwatch.elapsed(TimeUnit.NANOSECONDS);
+      long delayNanos = Math.max(0, retryBackoffPolicy.nextBackoffNanos() - elapsed);
+      logger.log(XdsLogLevel.INFO, "Retry ADS stream in {0} ns", delayNanos);
```

Comment:
We probably still want this log statement to be after the "ADS stream closed/failed with status" log statement below. We can easily move just this one line.

Reply:
Done